Lib/unittest/mock.py

```diff
@@ -1015,7 +1012,8 @@ def _get_child_mock(self, /, **kw):
         For non-callable mocks the callable variant will be used (rather than
         any custom subclass)."""
         _new_name = kw.get("_new_name")
-        if _new_name in self.__dict__['_spec_asyncs']:
+        _spec_val = getattr(self.__dict__["_spec_class"], _new_name, None)
+        if _spec_val is not None and asyncio.iscoroutinefunction(_spec_val):
             return AsyncMock(**kw)
 
         if self._mock_sealed:
```

Comment:
Not related to your PR, but not sure the behaviour of seal + async mocks is correct, will open an issue!